assimilation_code/modules/io/direct_netcdf_mod.f90

```diff
@@ -833,20 +848,27 @@ subroutine read_variables(ncfile_in, var_block, start_var, end_var, domain)
 integer,  intent(in)    :: end_var
 integer,  intent(in)    :: domain
 
+character(len=*), parameter :: routine = 'read_variables'
+
 integer :: i
 integer :: istart, iend
 integer :: var_size
 integer, allocatable :: dims(:)
 integer :: ret, var_id
+character(len=NF90_MAX_NAME) :: varname
+
+logical :: allow_missing
+logical :: failure
 
-logical :: missing_possible
+failure = .false.
 
-missing_possible = get_missing_ok_status()
+allow_missing = get_missing_ok_status()
```

Comment:
I think this variable name reads better.

Reply:
this was originally 'allow_missing' in much of the code.  but i remember changing the name because 'allow_missing' sounds like a verb and that you might be able to set it to change the behavior.  'missing_possible' sounded more like the result, not the action.  not a bit deal, but i'm not unhappy with the original name here ('missing_possible').